### ISSUE TITLE
Class-loading issue with test suite

### ISSUE THREAD

**Issue by quintesse** [ceylon/ceylon-compiler#2036]
The test suite, when run from the IDE, seems (in certain cases?) to load classes from the file system instead of the .car files. I noticed this when trying to run the test suite on Windows and getting a number of duplicate declaration errors. In each case it was a test that had both a Java class called `BugXXX` and a Ceylon method called `bugXXX` (see for example `bug850` and `bug854`).

**Comment by quintesse** [comment]
PS: Stef, I assigned this to you for a moment because I wanted to know if you had an idea. This could be as simple as a misconfigured IDE, but I'm not sure.

**Comment by FroMage** [comment]
Mmm, we do clear the classpath though, can you reproduce this?

**Comment by quintesse** [comment]
Yes , every time. But I can even reproduce the classpath problem on Linux, but because of the case sensitive paths it doesn't pose any problems. I just put a breakpoint in `CeylonModelLoader.lookupNewClassMirror()` around line 352 and look at the `classSymbol.classfile` and it shows things like: `RegularFileObject[.\build\classes\com\redhat\ceylon\compiler\java\test\issues\bug08xx\bug850\bug850.class]`

**Comment by FroMage** [comment]
So how does `build` end up in the classpath then?

**Comment by quintesse** [comment]
No idea, I was hoping you could tell me :)
Is there any place I can see that?
I also thought we had this fixed a long time ago, but I'm not entirely clear *how* it was fixed.

**Comment by FroMage** [comment]
See db0bee91780bb6cf9a43ebe09f68a0b356b90471

**Comment by quintesse** [comment]
@FroMage the result from `computeUserClassPath()` is :

```[..\ceylon-spec\bin,
 .\build\classes,
 .\..\ceylon-module-resolver\build\classes,
 .\..\ceylon-common\build\classes,
 .\..\ceylon-runtime\build\classes,
 .\..\ceylon-runtime\lib\jboss-modules-1.3.3.Final.jar,
 .\..\ceylon-module-resolver\lib\slf4j-api-1.6.1.jar,
 ..\ceylon.language\ide-dist\ceylon.language-1.1.1.car]```

And these are hard-coded here: https://github.com/ceylon/ceylon-compiler/blob/e3c09a5db3f4c65cae13a0ca3f1e8377b2152c81/test/src/com/redhat/ceylon/compiler/java/test/CompilerTest.java#L93

**Comment by quintesse** [comment]
Btw, deeper down the rabbit hole in the `JavacFileManager` the path has been extended to include more things:

```
[..\ceylon-spec\bin,
 .\build\classes,
 .\..\ceylon-module-resolver\build\classes,
 .\..\ceylon-common\build\classes,
 .\..\ceylon-runtime\build\classes,
 .\..\ceylon-runtime\lib\jboss-modules-1.3.3.Final.jar,
 .\..\ceylon-module-resolver\lib\slf4j-api-1.6.1.jar,
 ..\ceylon.language\ide-dist\ceylon.language-1.1.1.car,
 C:\Projects\ceylon\ceylon-compiler\build\test-cars\issues-0500-0999\com\redhat\ceylon\compiler\java\test\issues\1\com.redhat.ceylon.compiler.java.test.issues-1.car,
 C:\Users\tako\.ceylon\repo\com\redhat\ceylon\compiler\java\1.1.1\com.redhat.ceylon.compiler.java-1.1.1.jar,
 C:\Users\tako\.ceylon\repo\com\redhat\ceylon\compiler\js\1.1.1\com.redhat.ceylon.compiler.js-1.1.1.jar,
 C:\Users\tako\.ceylon\repo\com\redhat\ceylon\common\1.1.1\com.redhat.ceylon.common-1.1.1.jar,
 C:\Users\tako\.ceylon\repo\com\redhat\ceylon\module-resolver\1.1.1\com.redhat.ceylon.module-resolver-1.1.1.jar,
 C:\Users\tako\.ceylon\repo\com\redhat\ceylon\typechecker\1.1.1\com.redhat.ceylon.typechecker-1.1.1.jar,
 C:\Users\tako\.ceylon\repo\org\antlr\runtime\3.4\org.antlr.runtime-3.4.jar,
 C:\Users\tako\.ceylon\repo\org\jboss\modules\1.3.3.Final\org.jboss.modules-1.3.3.Final.jar,
 C:\Users\tako\.ceylon\repo\org\jboss\jandex\1.0.3.Final\org.jboss.jandex-1.0.3.Final.jar,
 C:\Users\tako\.ceylon\repo\ceylon\language\1.1.1\ceylon.language-1.1.1.car]
```

**Comment by quintesse** [comment]
Ok, got it, it's indeed an Eclipse configuration. It compiles the test files which end up in `.\build\classes` which is first in the class path. There's already a couple of bug folders that are in the exclusion list of `test/src`. Fixing...

**Comment by quintesse** [comment]
Joy, I'm actually encountering a whole bunch of tests that were actually depending on Eclipse to compile the .java files! (See `bug334` for example)

**Comment by quintesse** [comment]
Fixed by @c7603cd0319db5825dfde805fa80f607f652e556
